### ISSUE TITLE
IAM::Policy/PolicyName won't accept !Refs any more

### ISSUE THREAD

**Issue by stig** [aws-cloudformation/cfn-python-lint#1086]
*cfn-lint version: (`cfn-lint --version`)*

0.23.1

*Description of issue.*

There seems to be an issue with using `!Refs` for PolicyNames. This is (IMO) a regression from 0.22. The following minimal template works in 0.22, but fails with 0.23.1:

```yaml
AWSTemplateFormatVersion: "2010-09-09"
Resources:
  Bucket:
    Type: 'AWS::S3::Bucket'
  Group:
    Type: AWS::IAM::Group
  Policy:
    Type: 'AWS::IAM::Policy'
    Properties:
      PolicyName: !Ref Bucket
      Groups: [!Ref Group]
      PolicyDocument:
        Version: "2012-10-17"
        Statement: []
```

Substituting `!Refs Bucket` for `!Sub "${Bucket}"` (which AFAICS is equivalent) makes it pass:

```yaml
AWSTemplateFormatVersion: "2010-09-09"
Resources:
  Bucket:
    Type: 'AWS::S3::Bucket'
  Group:
    Type: AWS::IAM::Group
  Policy:
    Type: 'AWS::IAM::Policy'
    Properties:
      PolicyName: !Sub "${Bucket}"
      Groups: [!Ref Group]
      PolicyDocument:
        Version: "2012-10-17"
        Statement: []
```

```
$ cfn-lint fail.yaml pass.yaml
E3008 Property "PolicyName" has no valid Refs to Resources at Resources/Policy/Properties/PolicyName/Ref
fail.yaml:10:7

$ diff -u fail.yaml pass.yaml
--- fail.yaml	2019-08-06 16:40:40.000000000 +0100
+++ pass.yaml	2019-08-06 16:39:24.000000000 +0100
@@ -7,7 +7,7 @@
   Policy:
     Type: 'AWS::IAM::Policy'
     Properties:
-      PolicyName: !Ref Bucket
+      PolicyName: !Sub "${Bucket}"
       Groups: [!Ref Group]
       PolicyDocument:
         Version: "2012-10-17"
```

I see there have been some changes regarding PolicyName, but I haven't been able to pinpoint which commit introduced this change in behaviour.
